clouddriver-kubernetes/src/main/java/com/netflix/spinnaker/clouddriver/kubernetes/names/KubernetesNamerRegistry.java

```diff
@@ -0,0 +1,59 @@
+/*
+ * Copyright 2020 Google, LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package com.netflix.spinnaker.clouddriver.kubernetes.names;
+
+import static com.google.common.collect.ImmutableMap.toImmutableMap;
+
+import com.google.common.collect.ImmutableMap;
+import com.netflix.spinnaker.clouddriver.kubernetes.description.manifest.KubernetesManifest;
+import com.netflix.spinnaker.clouddriver.names.NamingStrategy;
+import com.netflix.spinnaker.kork.annotations.NonnullByDefault;
+import java.util.List;
+import java.util.Optional;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+/**
+ * This class handles registering any naming strategies for kubernetes manifests that are on the
+ * classpath, and supports looking these up by name. It is in principle possible for users to add
+ * additional namers in a custom build, but it is not clear how often this is used. The only namer
+ * that exists upstream is {@link KubernetesManifestNamer}.
```

Comment:
This feels like an unnecessary level of abstraction. Do you think we're ever going to register more namers? If we don't have any evidence that this is a useful extension point, feels like we should consider ripping out the registry and just using `KubernetesManifestNamer` as a static utility, similar to `KubernetesManifestAnnotator` or `KubernetesManifestLabeler`.

Reply:
Oh, I guess we [document](https://spinnaker.io/guides/developer/extending/crd-extensions/#custom-spinnaker-naming-strategies) this as an extension point...

Reply:
Yeah, I completely agree that the benefit of allowing this to be customized is unclear.  In particular, most of what `KubernetesManifestNamer` does is just apply and read annotations, which should work regardless of whether it is naming a built-in object or a CRD.  Maybe someone would want to change the default strategy for naming something that doesn't have the annotations applied (which defaults to frigga now)...but in that case we should probably just allow that to be extended rather than the whole namer.

I did consider just making it a static utility, but at least at this point didn't want to break the extensibility.  Let's discuss at the SIG and if we don't hear of anyone using it I'm on board with making it a static utility, which would greatly simplify things (and make it that much easier to create credentials in a test too).